arch/xtensa/core/ptables.c

```diff
@@ -327,6 +327,24 @@ void xtensa_mmu_init(void)
 	arch_xtensa_mmu_post_init(_current_cpu->id == 0);
 }
 
+void xtensa_mmu_reinit(void)
+{
+	/* First initialize the hardware */
+	xtensa_init_paging(xtensa_kernel_ptables);
+
+#ifdef CONFIG_USERSPACE
+	struct k_thread *thread = _current_cpu->current;
+	struct arch_mem_domain *domain =
+			&(thread->mem_domain_info.mem_domain->arch);
```

Comment:
What's the circumstance where this is happening?  Surely the only context where a core is coming back from a suspend state will be in the idle thread, no?  Why check _current?  Seems like needless complexity and an extra path for things to go wrong; shouldn't we just be restoring the kernel map always and relying on an upcoming context switch to figure out thread state?

Reply:
that is a good point, but right now when userpsace is enabled, if I not mistaken, idle threads are part of `k_mem_domain_default` that has a copy of kernel page tables (that could have changed). That is why I've done it. Thinking a little bit more about, it should be safe using `xtensa_kernel_ptables` for the idle thread, but to be fair that shouldn't cause much overhead.

Btw, I have another pr avoiding this page table duplication similarly with what X86 does.